modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/server/BasicOperationsKeyValueStorageTest.java

```diff
@@ -190,6 +191,88 @@ void getWithRevisionBound() {
         assertFalse(key3EntryBounded5.empty());
     }
 
+    @Test
+    void getWithRevisionLowerUpperBound() {
+        byte[] key1 = key(1);
+        byte[] key2 = key(2);
+
+        byte[] val1 = keyValue(1, 1);
+        byte[] val2 = keyValue(1, 2);
+        byte[] val3 = keyValue(2, 3);
+        byte[] val4 = keyValue(1, 4);
+        byte[] val5 = keyValue(1, 5);
+        byte[] val6 = keyValue(2, 6);
+        byte[] val7 = keyValue(2, 7);
+        byte[] val8 = keyValue(1, 8);
+        byte[] val9 = keyValue(1, 9);
+
+        assertEquals(0, storage.revision());
+        assertEquals(0, storage.updateCounter());
+
+        putToMs(key1, val1);
+        putToMs(key1, val2);
+        putToMs(key2, val3);
+        putToMs(key1, val4);
+        putToMs(key1, val5);
+        putToMs(key2, val6);
+        putToMs(key2, val7);
+        putToMs(key1, val8);
+        putToMs(key1, val9);
+
+        removeFromMs(key1);
+
+        assertEquals(10, storage.revision());
+        assertEquals(10, storage.updateCounter());
+
+        // Get entries with the lower revision and the upper revision are bound to the key.
+        // One entry from the revision range is bound to another key.
```

Comment:
It's still unclear.

Reply:
I've tried to describe it more clearly. Not sure if I succeeded. But I can't write better.